phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseRowKeyMatcherTestIT.java

```diff
@@ -504,7 +507,42 @@ private Pair<String, byte[]> getRowKeyMatchersFromView(PhoenixConnection connect
 
     }
 
-    // Helper to get rowKeyMatcher from Metadata.
+    byte[] getRowKeyMatcherFromSyscatIndex(String tenantId, String schemaName,
+            String tableName, boolean useIndexTable) throws SQLException {
+
+        final String
+                SYS_CATALOG_ROW_KEY_MATCHER_HEADER_SQL =
+                "SELECT ROW_KEY_MATCHER FROM SYSTEM.CATALOG " + "WHERE %s AND TABLE_SCHEM <> 'SYSTEM' AND TABLE_NAME = '%s' AND  ROW_KEY_MATCHER IS NOT NULL";
+        final String  SYS_CATALOG_IDX_ROW_KEY_MATCHER_HEADER_SQL = "SELECT \"0:ROW_KEY_MATCHER\" FROM SYSTEM.SYS_ROW_KEY_MATCHER_IDX " + "WHERE %s AND \":TABLE_SCHEM\" = '%s' AND \":TABLE_NAME\" = '%s'";
```

Comment:
Here you are explicitly querying the index. Are you asserting somewhere where we query syscat but the optimizer transparently generates the plan for index and reads from index.

Reply:
I have some [test](https://github.com/apache/phoenix/blob/da47478500b085221a22b05e4e1b73e23bd6dbc1/phoenix-core/src/it/java/org/apache/phoenix/end2end/index/PartialSystemCatalogIndexIT.java#L825) in PartialSystemCatalogIT - which test the explain plan generated